lib/stub_ui/lib/src/ui/window.dart

```diff
@@ -35,6 +35,71 @@ typedef PlatformMessageResponseCallback = void Function(ByteData data);
 typedef PlatformMessageCallback = void Function(
     String name, ByteData data, PlatformMessageResponseCallback callback);
 
+/// Time-related performance metrics of a frame.
+class FrameTiming {
+  const FrameTiming._(this.buildStart, this.buildFinish, this.rasterStart, this.rasterFinish);
+
+  /// The time that UI thread starts building this frame.
+  ///
+  /// {@template dart.ui.FrameTiming.epoch}
+  /// This is a raw timestamp from some epoch. The epoch in all [FrameTiming] is
+  /// the same, but it may not match [DateTime]'s epoch.
+  /// {@endtemplate}
+  final Duration buildStart;
+
+  /// The time that UI thread finishes building this frame.
+  ///
+  /// {@macro dart.ui.FrameTiming.epoch}
+  final Duration buildFinish;
+
+  /// The time that GPU thread starts rasterizing this frame.
+  ///
+  /// {@macro dart.ui.FrameTiming.epoch}
+  final Duration rasterStart;
+
+  /// The time that GPU thread finishes rasterizing this frame.
+  ///
+  /// {@macro dart.ui.FrameTiming.epoch}
+  final Duration rasterFinish;
+
+  /// The time to build the frame on the UI thread.
+  ///
+  /// {@template dart.ui.FrameTiming.fps_smoothness_milliseconds}
+  /// To ensure smooth animations of K fps, this should not exceed 1000/K milliseconds.
```

Comment:
maybe use a green letter to make it clear that K is a variable and not the SI prefix for Kilo?

Reply:
my bad, i meant "greek", not green!